src/cdk/table/table.ts

```diff
@@ -393,6 +439,56 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
     this._customRowDefs.delete(rowDef);
   }
 
+  /**
+   * Get the list of DataRow objects to render according to the current list of data and defined
+   * row definitions. If the previous list already contained a particular pair, it should be reused
+   * so that the differ equates their references.
+   */
+  private _getDataRows(): DataRow<T>[] {
```

Comment:
Having both `_getDataRows` and `_getDataRowsForData` is confusing

Reply:
Agreed, struggled coming up with a better name for these. One returns the list of all the `DataRow` objects to render, while one returns the list of `DataRow` objects for a particular data object.

Any ideas?

Reply:
I think `_getAllRenderRows` and `_getRenderRowsForData` would be good